[CS2103T-F13-3] MoneyGoWhere

Nanosync opened the pull request:
@austinsantoso
@choongyx
@jonathantjendana
@minpyaemoe

halfwhole commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|=======================================================================
+|`* * *` |user |see the difference between my budget and spending|be aware of my finances
+
+|`* * *` |new user |view more information about a particular command|learn how to various commands
```
Did you mean: learn how to use various commands?

halfwhole commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |user |redo my previous action | move forward in history
+
+|`* *` |financial analyst |view daily/monthly/yearly expenditure |have a projected view of all my spending
```
Who is a financial analyst? Is it a part of your target audience?

LiuZechu commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
+
+'''
+[discrete]
+=== UC12 Graphing the data
```
You may consider specifying the range of data used for the graph e.g. a month/year/etc.

halfwhole commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |forgetful user |set a reminder |reminded of my payment deadlines
+
+|`*` |housewife |view the cheapest price of an item |improve my savings
```
Why would only _housewives_ want to view the cheapest price of an item? I'm sure others would like to use this feature too.

halfwhole commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-_{More to be added}_
+|`*` |user who loves rewards |see how much more I should spend |earn rewards from places I spend at
+
+|`*` |investor |see how much I should invest | earn more money
```
What does `how much I should invest` mean in the context of your app?

LiuZechu commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
+
+*MSS*
+
+1.  User requests to clear all entries in the list of spending.
```
The system should ask the user for confirmation before deleting everything.

halfwhole commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |user |view my current monthly spending on different categories|keep track of my spending
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |budget conscious user |set budget goals|make sure I do not overspend
```
budget-conscious

LiuZechu commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1.  User requests to view the list of spending.
+2.  MoneyGoWhere shows a list of spending.
+3.  User requests to delete a specific spending from the list.
```
The system should ask the user for confirmation before deleting.

uggi121 commented on docs/UserGuide.adoc:
```diff
@@ -12,166 +12,288 @@ ifdef::env-github[]
+* Prefixes
+** day after/ the day before
+** the monday after/ the monday before
+** 2 fridays before/ 4 tuesdays after
```
Quite exhaustive! Looks good

uggi121 reviewed (commented):
Initial outlook is very positive!

uggi121 commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,92 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |calculate my total spending |keep track of my total spending
+
+|`* *` |user |import data |have different data from external source
```
import what data?

uggi121 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
+.  Should not require user to install.
+.  Features implemented should be testable using automated and manual testing.
+.  Should work for a single user only.
+.  Should be able to run with or without internet connection.
```
how do you get information like exchange rates without the internet?

lyskevin commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
+*MSS*
+
+1.  User requests to import data from CSV file by specifying the file path.
+2.  MoneyGoWhere imports the data from the CSV file located at specified file path.
```
"file path" might be a bit too technical for a use case. Maybe the specification could be worded as "User requests to import data from a CSV file" instead.

lyskevin commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +397,232 @@ Use case ends.
+
+.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
+.  Should be able to hold up to 1000 spending without a noticeable sluggishness in performance for typical usage.
+.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
It is not clear how this NFR would be accomplished. Perhaps an example of a benchmark can be given (e.g. adding an item in 3 seconds when using the keyboard compared to 5 seconds with the GUI).

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+The `StatsCommandParser` differentiates these 2 commands based on whether a valid date range is provided as illustrated in the diagram below.
+
+.Activity Diagram showing behaviour of StatsCommandParser
+image::statsactivitydiagram.png[width="600"]
```
Is something wrong with the link? Does not load for us.

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+The following sequence diagram shows how `stats d/01/01/2019 d/07/01/2019` works during the preparation phase.
+
+.Sequence Diagram demonstrating the preparation stage of Statistics feature
+image::StatsSequenceDiagram.png[]
```
Should the lifeline of StatsCommandParser continue after the cross?

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+Below is the UML sequence diagram and a step-by-step explanation for the retrieval of `statsData`. `statsMessage` is retrieved in the same manner.
+
+.Sequence Diagram of Execution Phase showing retrieving of `statsData`
+image::StatsMessageSequenceDiagram.png[]
```
Should True in the call be `true` with the lowercase instead?

Should the lifeline of StatsCommandParser continue after the cross?

Should Model have an activation box?

Should the return arrows be dotted lines instead?

Is CommandResult meant to be there? It seems to have no interaction.

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+.Sequence Diagram of Execution Phase showing retrieving of `statsData`
+image::StatsMessageSequenceDiagram.png[]
+
+1. Upon checking that the boolean `isShowStats` = true, the MainWindow calls `LogicManager#getStatsData()`.
```
Is there a method that checks this boolean? If there is, the top left arrow in the previous diagram should be that method as well.

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+
+===== How to pass statistics data parameters from Logic to UI
+
+Alternative 1 (current choice): UI gets all data and parameters from Logic, which gets data from Model.
```
Perhaps you could bold or somehow show that these subsequent pros and cons are for these alternatives?

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+To summarise what happens when the user uses the `find` command, the following activity diagram is shown below:
+
+.Activity diagram for an example `find` command
+image::FindActivityDiagram.png[Find Activity Diagram,width=75%]
```
What is List<Predicate<Spending>>? Should this be a class? Not clear what it's used for apart from being constructed.

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+.. If there is an invalid field, display an error message for the invalid field.
+.. If there is a missing field, display an error message for the missing field.
+
+==== Design Considerations
```
This format is good. Perhaps the rest of the Dev Guide can follow the same format?

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+The sequence diagram below demonstrates how the `sort` command is executed:
+
+.Sequence diagram for an example `sort` command
+image::SortSequenceDiagram.png[Sort Sequence Diagram,width=75%]
```
Should the `fields` call to `SortCommand` be the actual constructor instead?

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+[NOTE]
+In the above diagram, it can be seen that respective error messages will be shown for invalid inputs.
+
+Shown below is the sequence diagram containing the interactions between respective components in MoneyGoWHere when user inputs `AddReminderCommand`.
```
Typo in MoneyGoWHere.

yaaanch commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+
+With this reminder feature, users can set the reminders of their tasks with deadlines and delete them when completed.
+
+==== Implementation
```
We think it would be most helpful if we could know how a reminder will remind the user.

brebeek commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,489 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+The sequence diagram below demonstrates how the `find` command is executed:
+
+.Sequence diagram for an example `find` command
+image::FindSequenceDiagram.png[Find Sequence Diagram,width=75%]
```
![image](https://user-images.githubusercontent.com/54146425/67548143-a67b4500-f733-11e9-86aa-ed046bb982f6.png)
May want to include a similar disclaimer:
![image](https://user-images.githubusercontent.com/54146425/67548177-c27ee680-f733-11e9-9a84-161ce90d22cb.png)
for FindCommand

brebeek commented on docs/DeveloperGuide.adoc:
```diff
@@ -92,15 +161,18 @@ The `UI` component,
+The following class diagram shows the structure of the Logic component.
+
 .Structure of the Logic Component
 image::LogicClassDiagram.png[]
```
Nice use of class diagram & clear elaboration. Good work

brebeek reviewed (commented):
Good job in updating most of the documentation. Clear and concise.

uggi121 commented on docs/DeveloperGuide.adoc:
```diff
@@ -114,99 +186,489 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+Cons: Not in alignment with architecture. Need to consider application startup when there are no events posted.
+
+Alternative 1 was chosen due to the ease of implementation.
+As the current statistics feature does not require a large amount of data, the number of method calls is not too large.
```
Doesn't this design implementation compromise scalability?

uggi121 reviewed (commented):
Nice job!

jun-ha0 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +832,302 @@ Use case ends.
+
+*Extensions*
+
+[none]
```
It might be possible to change the extension step from 1a. and 1a1. to 2a. and 2a1. respectively.

jun-ha0 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +832,302 @@ Use case ends.
++
+Use case resumes at step 1.
+[none]
+
```
It might be possible to change the extension step from 1a. and 1a1. to 2a. and 2a1. respectively.

jun-ha0 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,46 +832,302 @@ Use case ends.
++
+Use case resumes at step 1.
+[none]
 
```
It might be possible to change the extension step from 1a. and 1a1. to 2a. and 2a1. respectively.

uggi121 commented on src/main/java/seedu/moneygowhere/logic/Logic.java:
```diff
@@ -0,0 +1,77 @@
+     * @return the hashmap of spending data
+     * @throws ParseException If an error occurs during parsing.
+     */
+    Map<Date, Double> getGraphData(String commandText) throws ParseException;
```
Alas! Thy design consideration to chooseth a wrapper class shall dilapidate thee temporal aspects of the program. The sheer ignominy and inefficacy of repeated autoboxing and unboxing can be averted with a steadfast class instead.

uggi121 reviewed (commented):
Thy product haveth immense potential and never have things been of more moment

minpyaemoe committed:
Clear command will now clear all the entries including reminders

minpyaemoe committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-3/main into branch-UG-Clear-Reminders

minpyaemoe committed:
added bug fix empty string reminder message, changed reminder date description,

minpyaemoe committed:
amened UG for clear command and addreminder commands

minpyaemoe committed:
added space regrex and update isValidname function

choongyx committed:
Update Stats and Graph to change with currency

choongyx committed:
Update incorrect command examples for add

choongyx committed:
Update invalid cost message

choongyx committed:
Correct command syntax for edit

choongyx committed:
Update Quickstart with information regarding preloaded data

choongyx committed:
Update UG for graph and stats

choongyx committed:
Remove unused Graphchart.java

choongyx committed:
Reposense Authorship

choongyx committed:
Remove unused code

jonathantjendana committed:
Fix importing empty tags

choongyx committed:
Changes based on review

jonathantjendana committed:
Added checks for various invalid csv files

jonathantjendana committed:
Added javadoc

jonathantjendana committed:
Refactored methods

jonathantjendana committed:
Updated UI

jonathantjendana committed:
Removed unused imports

choongyx committed:
Merge pull request #205 from choongyx/FinalSubmission

Address Bug Reports from Practical Exam Dryrun

jonathantjendana committed:
Removed unused import

minpyaemoe committed:
Amended UG accoridng to review and bug fix for same spending notification, help command

minpyaemoe committed:
Amended user guide accoridng to review

minpyaemoe committed:
Added 1 year deadline behind check for adding reminder

minpyaemoe committed:
fix checkstyle issue

Nanosync committed:
Merge pull request #208 from jonathantjendana/branch-update-ui

Updated UI

minpyaemoe committed:
Added PPP

minpyaemoe committed:
changed UG for add reminders exactly like what reviewer said

minpyaemoe committed:
Merge pull request #203 from minpyaemoe/branch-Clear-Reminders-and-bug-fixes

Added clearing reminder, bug fix for #190 and #163, and changed date description for reminder deadline

Nanosync committed:
Merge pull request #207 from jonathantjendana/branch-import-bugfix

Bug Fix for Import

austinsantoso committed:
Documentation for updown

added documentation for up down

austinsantoso committed:
fix styling

fix styling

Nanosync committed:
Fix sort command with repeated fields

Nanosync committed:
Fix sort predicates

Nanosync committed:
Update spending comparator test based on new sort command fixes

Nanosync committed:
Refactor currency command

Nanosync committed:
Annotate DateUtil for RepoSense

Nanosync committed:
Fix find command and update UG

Nanosync committed:
Apply reposense to relevant code sections

Nanosync committed:
Update DateUtil and range of dates (1y for reminder, 1m for spending)

Nanosync committed:
Update exchanger rate disclaimer

Nanosync committed:
Updated add and edit command to use new currency, fix UG summary

Nanosync committed:
Update reposense to ignore some commits

Nanosync committed:
Codacy fix

Nanosync committed:
Merge pull request #210 from austinsantoso/branch-updownugdg

Documentation for updown

Nanosync committed:
Merge pull request #211 from Nanosync/branch-find

Address PE Dry Run bugs

austinsantoso committed:
rework how budget calculates sum

rework how budget calculates sum, now reads file. created jsonAdaptedBudget test 
fix budget not changing when edit command is used

austinsantoso committed:
styling for dg

fix styling for dg

Nanosync committed:
Fix test names

Nanosync committed:
Add reposense to tests

Nanosync committed:
Merge pull request #215 from austinsantoso/branch-fixupdownugdg

styling for dg

Nanosync committed:
Merge pull request #216 from Nanosync/branch-cq

Fix test names

Nanosync committed:
Merge pull request #214 from austinsantoso/branch-budgetReadStorage

rework how budget calculates sum

austinsantoso committed:
add text to Budget Panel

add text 'Expenditure' to Budget Panel

austinsantoso committed:
change text from 'expenditure' to 'remaining'

change text from 'expenditure' to 'remaining'

minpyaemoe committed:
added code ownership and javadoc comments

minpyaemoe committed:
resolved merge conflict

minpyaemoe committed:
ammeded respective integration test

minpyaemoe committed:
Merge branch 'branch-same-spending-notification' of https://github.com/minpyaemoe/main into branch-PPP

Nanosync committed:
Update sample data

austinsantoso committed:
add budget limit, change error message

revert previous commit
budget still can only have negative value
Budget has limit of 1000000000
changed error message
remove getDateValue() from Date

Nanosync committed:
Merge pull request #220 from Nanosync/branch-data

Update sample data

Nanosync committed:
Merge pull request #217 from austinsantoso/branch-budgetUi

add text to Budget Panel

Nanosync committed:
Merge pull request #219 from minpyaemoe/branch-same-spending-notification

Added notification for same spending, amended help command, fixes #199 and #181

Nanosync committed:
Merge pull request #218 from austinsantoso/branch-budgetLimit

Bug fixes, rework Budget

Nanosync committed:
Change sorting order again

Nanosync committed:
Merge pull request #221 from Nanosync/branch-sort

Change sorting order again

jonathantjendana committed:
allow mass addition to spending list rather than adding one spending at a time

jonathantjendana committed:
update spending list card horizontal width

jonathantjendana committed:
Updated UI Screenshot

jonathantjendana committed:
updated ui_annotate.png

jonathantjendana committed:
Merge pull request #224 from jonathantjendana/branch-update-import

Updated import & minor change for Ui

austinsantoso committed:
integrate currency with budget

Budget now changes with currency

jonathantjendana committed:
Merge pull request #223 from jonathantjendana/branch-update-ss

Updated UI Screenshot

austinsantoso committed:
fix to JUnit standard

rename setup to setUp

Nanosync committed:
Merge pull request #225 from austinsantoso/branch-currencyForBudget

integrate currency with budget

Nanosync committed:
Modify UG to add notice for different currency when adding spending

Nanosync committed:
Merge pull request #226 from Nanosync/branch-update-docs

Modify UG to add notice for different currency when adding spending

jonathantjendana committed:
Updated UG & DG

jonathantjendana committed:
Replaced MainWindow.fxml with MainWindow2.fxml

choongyx committed:
Merge pull request #229 from jonathantjendana/branch-deleted-mainwindow2

Replaced MainWindow.fxml with MainWindow2.fxml

choongyx committed:
Merge pull request #228 from jonathantjendana/branch-update-guides

Updated DG

choongyx committed:
Fix graph predicate bug

Nanosync committed:
Added NLP fixes

Nanosync committed:
Merge pull request #232 from choongyx/FinalSubmission

Fix graph predicate bug

choongyx committed:
Fix another bug related to graph and stats not refreshing

jonathantjendana committed:
Merge pull request #233 from Nanosync/branch-nlp

Added NLP fixes

Nanosync committed:
Add natty to readme

Nanosync committed:
Merge pull request #234 from choongyx/FinalSubmission

Fix another bug related to graph and stats not refreshing

jonathantjendana committed:
format large numbers

Nanosync committed:
Add currency to status bar

jonathantjendana committed:
Added test cases & format stats data

jonathantjendana committed:
added reposense author tag

Nanosync committed:
Merge pull request #237 from Nanosync/branch-nlp

Add natty to readme

Nanosync committed:
Merge pull request #236 from Nanosync/branch-exchange-rates

Add currency to status bar

Nanosync committed:
Merge pull request #235 from jonathantjendana/branch-update-largenumberformat

Format Large Numbers

choongyx committed:
Limit tags to 15 characters

choongyx committed:
Add date range to graph and stats label

minpyaemoe committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-3/main into branch-PPP

choongyx committed:
Resolve bug related to graph and stats fixed view

minpyaemoe committed:
added code reuse description for reminder comparator, intergration test for duplicate spending with changed currency and changed duplicate spending notification message

minpyaemoe committed:
aded regex for brackets, fix reminder message text wrapping bug

choongyx committed:
Code Cleanup, Remove old code, JavaDocs

choongyx committed:
Test cleanup

choongyx committed:
Fix checkstyle errors

minpyaemoe committed:
added enum for reminder type

choongyx committed:
Fix checkstyle errors

Nanosync committed:
Merge pull request #241 from choongyx/FinalSubmission

Limit tags to 15 char, Update graph and stats, Cleanup

Nanosync committed:
Update sample data for sorting

choongyx committed:
Update UG and DG

Nanosync committed:
Merge pull request #242 from Nanosync/branch-sort

Update sample data for sorting

choongyx committed:
Update DG, including sections DevOps, Testing, Documentation etc

choongyx committed:
Update DG and PPP

choongyx committed:
Remove html file

choongyx committed:
Typo

jonathantjendana committed:
- Update UG, DG & PPP
- Added logging for import & export
- Changed budget remark colour
- Changed Ui screenshot

austinsantoso committed:
ug,dg and ppp for my section

Added ug, dg and ppp for sections related to me

austinsantoso committed:
prevent user from entering `budget 0.000000001`

prevent user from entering `budget 0.000000001`, but making sure the input is not less than 0.01

choongyx committed:
Remove more unused code

choongyx committed:
Update DG diagrams based on prior update

choongyx committed:
Update DG based on reviews

choongyx committed:
Resolve errors with regards to tests

choongyx committed:
Update incorrect links in PPP

Nanosync committed:
Fix more NLP bugs

Nanosync committed:
Fix typo in attribution for reposense

choongyx committed:
Remove comment from DG

Nanosync committed:
Update to PPP, DG and UG

Nanosync committed:
Merge pull request #244 from choongyx/FinalSubmission

Update Documentations, including tag limit

Nanosync committed:
Merge pull request #246 from austinsantoso/branch-uddgppp

Added UG, DG and PPP (Austin)

Nanosync committed:
Merge pull request #247 from austinsantoso/branch-budgetfix

prevent user from entering `budget 0.000000001`

Nanosync committed:
Merge pull request #248 from Nanosync/branch-nlp

Fix NLP bugs again

Nanosync committed:
Merge branch 'master' of github.com:AY1920S1-CS2103T-F13-3/main into branch-ppp

# Conflicts:
#	docs/UserGuide.adoc

Nanosync committed:
Add more PPP updates

Nanosync committed:
Merge pull request #249 from Nanosync/branch-ppp

Update PPP, DG and UG (Benedict)

Nanosync committed:
Fix errors in PPP

Nanosync committed:
Merge pull request #250 from Nanosync/branch-ppp

Fix errors in PPP (Benedict)

minpyaemoe committed:
update PPP, usage of enum for reminder type and added test cases for reminder type

minpyaemoe committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-3/main into branch-PPP-bracket-regex-UI-bug-fix

minpyaemoe committed:
Updated PPP with DG

minpyaemoe committed:
CheckStyle Fix

minpyaemoe committed:
added new line at PPP

choongyx committed:
Resolve minor mistakes in documentations

choongyx committed:
Remove html file

minpyaemoe committed:
added link for PPP and update photo

choongyx committed:
Correct typo mistake

choongyx committed:
Fixes #253

jonathantjendana committed:
Update UG

jonathantjendana committed:
updated ui annotation

jonathantjendana committed:
Fixed test case

jonathantjendana committed:
Changed UI screenshot

jonathantjendana committed:
typo

Nanosync committed:
Update default sorting order

Nanosync committed:
Rename test cases

Nanosync committed:
Force asciidoc to centre figure captions and hide links in print

Nanosync committed:
Update PPP, DG and UG

Nanosync committed:
Merge pull request #252 from choongyx/FinalSubmission

[Yong Xin] Resolve minor mistakes in documentations

Nanosync committed:
Merge pull request #251 from minpyaemoe/branch-PPP-bracket-regex-UI-bug-fix

Add (MIN PYAE MOE) PPP, solution for remaining bug fixes

Nanosync committed:
Merge pull request #254 from jonathantjendana/branch-update-ppp

Update PPP, UG, DG

Nanosync committed:
Update sort order for UG and DG

Nanosync committed:
Cleanup label for sort preview

Nanosync committed:
Merge pull request #255 from Nanosync/branch-ppp

Fix PPP (Benedict) and rename test cases

Nanosync committed:
Hide URLs and center images in a failsafe way

Nanosync committed:
Merge pull request #256 from Nanosync/branch-ppp

Hide URLs and center images in a failsafe way

Nanosync committed:
Cleanup UG again

Nanosync committed:
Merge pull request #257 from Nanosync/branch-ppp

Cleanup UG (Benedict)

Nanosync committed:
Fix DG reference

Nanosync committed:
Merge pull request #258 from Nanosync/branch-ppp

Final PPP

minpyaemoe committed:
limited the deadline of a reminder to be only in the future before the end of next year
and updated UG for adding tag and removing tag

minpyaemoe committed:
amended isSameSpending method and removed invalid testcases

minpyaemoe committed:
removed left over words "address" from project

Nanosync committed:
Add test cases for currency and exchangerate in DG

Nanosync committed:
Merge pull request #260 from Nanosync/branch-ppp

Add test cases for currency and exchangerate in DG

Nanosync committed:
Merge pull request #259 from minpyaemoe/branch-update-UG-for-adding-tag-removing-tag

Updating UG for add/remove tag, reminder deadline constraint and amended isSameSpending method

austinsantoso committed:
Add ppp to aboutUs page

Add ppp to aboutUs page

choongyx committed:
Merge pull request #262 from austinsantoso/branch-ugdgppp

Add ppp to aboutUs page

choongyx committed:
Update DG manual testing

austinsantoso committed:
fix ppp and dg

fix ppp and dg

Nanosync committed:
Merge pull request #263 from choongyx/Documentation

Update DG manual testing

choongyx committed:
fix cost = 0 bug

Nanosync committed:
Merge pull request #264 from austinsantoso/branch-ugdgppp

fix ppp and dg

Nanosync committed:
Merge pull request #265 from choongyx/master

fix cost = 0 bug

austinsantoso committed:
implement check decimal point

with also checking with trailing zero

choongyx committed:
Remove full stops

austinsantoso committed:
fix design of budget

fix design of budget, add 2 dp restriction for budget

austinsantoso committed:
Merge pull request #266 from choongyx/branch-PPP

Update minor typos in PPP

choongyx committed:
Update incorrect links in PPP

minpyaemoe committed:
Update DG, Logic Class diagram, Model Class diagram

minpyaemoe committed:
added highlight for class name and EOF in the diagram

minpyaemoe committed:
removed reminderlist from reminder class diagram, added ReminderCommandParser Component

minpyaemoe committed:
redrew logic class diagram

minpyaemoe committed:
Merge pull request #268 from choongyx/branch-PPP

Update incorrect links in PPP

minpyaemoe committed:
added architectureSequence Diagram for adding reminder, Design consideration into DG

minpyaemoe committed:
added extra for checkstyle

jonathantjendana committed:
compress the Summary of My Contributions

jonathantjendana committed:
Commented off component contributions

minpyaemoe committed:
amended according to the code review

jonathantjendana committed:
moved purpose section on top of format section for Import & Export command

jonathantjendana committed:
Changed .csv to Csv

austinsantoso committed:
Merge pull request #267 from austinsantoso/branch-bugfix

add 2 dp restriction for budget

minpyaemoe committed:
changed "user" to "you"

jonathantjendana committed:
Split the summary of contributions

minpyaemoe committed:
Amended UserGuide and PPP

Nanosync committed:
Add loggers

Nanosync committed:
Updated PPP phrasing

minpyaemoe committed:
adjusted spaces for PPP

austinsantoso committed:
grammar fix

change user to you in UG, some grammar fix

Nanosync committed:
Delete extra line in DG

Nanosync committed:
Merge branch 'master' of github.com:AY1920S1-CS2103T-F13-3/main into branch-ppp

# Conflicts:
#	docs/DeveloperGuide.adoc

Nanosync committed:
Remove test case line

Nanosync committed:
Remove bolded text in PPP for consistency

austinsantoso committed:
revert "user" to "you"

revert "user" to "you"

minpyaemoe committed:
updated UG and DG according to review

jonathantjendana committed:
- deleted testing for saving data
- updated grammar for import in UG

Nanosync committed:
Merge pull request #271 from Nanosync/branch-ppp

Added loggers and added credits

minpyaemoe committed:
adding test cases

minpyaemoe committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-3/main into branch-update-DG

jonathantjendana committed:
Fixed typos

minpyaemoe committed:
added test cases for adding and removing reminder in DG

minpyaemoe committed:
Merge pull request #269 from minpyaemoe/branch-update-DG

Updates for DG, Logic Class diagram, Model Class diagram

jonathantjendana committed:
Merge branch 'master' into branch-update-ppp

Nanosync committed:
Updated DG, version and PPP

jonathantjendana committed:
Merge pull request #270 from jonathantjendana/branch-update-ppp

Update PPP

minpyaemoe committed:
adjusting the table

Nanosync committed:
Merge pull request #273 from Nanosync/branch-update-docs

Updated DG, version and PPP

austinsantoso committed:
fix grammar and typos

fix grammar and typos for the last time i hope

minpyaemoe committed:
changing the F to f

austinsantoso committed:
fix commandHistory sequence diagram

fix commandHistory sequence diagram

austinsantoso committed:
fix styling

fix styling

austinsantoso committed:
remove random semicolon

remove random semi colon

minpyaemoe committed:
captalized s

minpyaemoe committed:
changed to the above table

minpyaemoe committed:
Merge branch 'master' into branch-update-DG

minpyaemoe committed:
csv to Csv

minpyaemoe committed:
Merge branch 'branch-update-DG' of https://github.com/minpyaemoe/main into branch-update-DG

austinsantoso committed:
Merge pull request #272 from austinsantoso/branch-ugusertoyou

grammar fix

jonathantjendana committed:
update PPP Overview

Nanosync committed:
Merge pull request #274 from minpyaemoe/branch-update-DG

Adjusting the Design Consideration table and adding tag

Nanosync committed:
Merge pull request #275 from jonathantjendana/branch-update-ppp-fix

Fix PPP Overview

Nanosync committed:
Update PPP text

austinsantoso committed:
add links

add links

minpyaemoe committed:
target at to "to"

Nanosync committed:
Merge pull request #276 from Nanosync/branch-update-docs

Update PPP text

Nanosync committed:
Merge pull request #277 from austinsantoso/branch-changereadme

add links

minpyaemoe committed:
resolved merge conflict

minpyaemoe committed:
budge to budget

Nanosync committed:
Update PPP text again

Nanosync committed:
Merge pull request #279 from Nanosync/branch-update-docs

Update PPP text again

Nanosync committed:
Merge pull request #278 from minpyaemoe/branch-update-DG

Fix consistency of UG, DG and Readme
